imports/plugins/core/accounts/server/no-meteor/mutations/inviteShopMember.js

```diff
@@ -0,0 +1,159 @@
+  const {
+    email,
+    groupId,
+    name,
+    shopId
+  } = input;
+
+  if (!userHasPermission(["reaction-accounts"], shopId)) {
+    throw new ReactionError("access-denied", "Access denied");
+  }
+
+  // we always use primary shop data, so retrieve this shop first with `Reaction` helper,
+  // and only query the `Shops` collection if shopId !== primaryShop._id
+  const primaryShop = Reaction.getPrimaryShop();
+  if (!primaryShop) throw new ReactionError("not-found", "No primary shop found");
+
+  let shop = primaryShop;
+  if (primaryShop._id !== shopId) {
+    shop = await Shops.findOne({ _id: shopId });
+  }
+  if (!shop) throw new ReactionError("not-found", "No shop found");
+
+  const group = await Groups.findOne({ _id: groupId });
+  if (!group) throw new ReactionError("not-found", "No gr found");
+
+  // we don't allow direct invitation of "owners", throw an error if that is the group
+  const ownerGroup = await Groups.findOne({ name: "owner" });
+  if (group._id === ownerGroup._id) {
+    throw new ReactionError("bad-request", "Cannot directly invite owner");
+  }
+
+  // An account can add another account to a group as long as the person adding
+  // has all permissions granted by that group.
+  // We can't use `userHasPermission` here because we want to make sure they
+  // have ALL the permissions rather than ANY.
+  // Accounts in the "Owner" group are able to add any user to any group,
+  // regardless of other permissions.
+  const contextUserAccount = await Accounts.findOne({ _id: userIdFromContext._id });
+  const isOwnerAccount = !!ownerGroup && contextUserAccount.groups.includes(ownerGroup._id);
+
+  const { permissions: groupPermissions = [] } = group;
+
+  if (!context.isInternalCall && !isOwnerAccount && difference(groupPermissions, userFromContext.roles[shopId] || []).length > 0) {
+    throw new ReactionError("access-denied", "Access Denied");
+  }
+
+  // get currentUserName from contextUserAccount to pass name to invite email
+  const currentUserName = getCurrentUserName(contextUserAccount); // TODO: make sure this works as it should
+
+  // check to see if invited user has an account
+  const invitedUser = await users.findOne({ "emails.address": email });
+
+  // if there is a userAccount, check to see if email has been verified
+  const matchingEmail = invitedUser && invitedUser.emails && invitedUser.emails.find((accountEmail) => accountEmail.address === email);
+  const isEmailVerified = matchingEmail && matchingEmail.verified;
+
+  // set variables to pass to email templates
+  let dataForEmail;
+  let userId;
+  let templateName;
+
+  if (invitedUser) {
+    userId = invitedUser._id;
+  }
+
+  // if the user already has an account, send in informative email instead of an invite email
+  if (invitedUser && isEmailVerified) {
+    // make sure user has an `Account` - this should always be the case
+    const invitedAccount = await Accounts.findOne({ _id: userId });
```

Comment:
Should be `Accounts.findOne({ userId });`. `account.userId` and `account._id` usually match but nothing says they have to if accounts were imported or something.

Reply:
This is confusing with all the users / accounts here. I think I do want `_id: userId` here, because at this point, `userId === user._id`, as set here: https://github.com/reactioncommerce/reaction/blob/06d7e7d7d1837e66615af710b14890c18226b501/imports/plugins/core/accounts/server/no-meteor/mutations/inviteShopMember.js#L85

Don't `user._id` and `Accounts._id` always match, even if `Accounts._id` and `Accounts.userId` don't?

Reply:
Nevermind on this, I see how I was misunderstanding what you were saying.